### ISSUE TITLE
ant deploy fails on buildCLI, following vagrant workflow

### ISSUE THREAD

**Issue by csantanapr** [openwhisk/openwhisk#331]
I'm not sure if this is a docker timeout problem, that needs to be increase more than 240s or that the docker host is dead

```
ant deploy
```

Error here
```
...
deploySlaves:
     [exec] The invoker hosts are [(0, '172.17.0.1', False)]
     [exec] Removing invoker0        at 172.17.0.1:4243     0.1 sec   did nothing
     [exec] Skipping pull
     [exec] Starting invoker0        at 172.17.0.1:4243     0.8 sec   SUCCEEDED
     [exec] Container 3f4477a118fb003a at 172.17.0.1:4243     0.0 sec   SUCCEEDED
     [exec]   has imageId   sha256:f95ca34f5978089680f9f3c7b8ad43adf1c10d63ec8ee1befb3105bb471f4a4e
     [exec] Checking invoker0                                               15.3 sec   SUCCEEDED
     [exec] 172.17.0.1:12001
     [exec] invoker0 is alive
     [exec]
     [exec] Tried to deploy 1 invokers. All done in 16.2 sec.

waitController:

waitComponent:
     [exec] controller is alive

waitActivator:

waitComponent:
     [exec] activator is alive

buildCLI:
    [mkdir] Created dir: /home/vagrant/openwhisk/build/tools/cli
     [copy] Copying 6 files to /home/vagrant/openwhisk/build/tools/cli
     [copy] Copying 40 files to /home/vagrant/openwhisk/build/tools/cli/openwhisk
     [move] Moving 1 file to /home/vagrant/openwhisk/build/tools/cli/openwhisk
     [move] Moving 1 file to /home/vagrant/openwhisk/build/tools/cli/openwhisk
     [exec] Wed May  4 19:04:50 UTC 2016
     [exec] With timeout 240, running:  docker  "build" "-t" "whisk/cli" "/home/vagrant/openwhisk/build/tools/cli"
     [exec] Cannot connect to the Docker daemon. Is the docker daemon running on this host?
     [exec] docker process has pid 12321
     [exec] 12321 pts/0    00:00:00 bash
     [exec] 12321 is still in the ps output. Must still be running.
     [exec] Docker process 12321 finished in time with status 1.
     [exec] Sleeping for 60 seconds to allow recovery (even though it wasn't killed)
     [exec] Done sleeping
     [exec] Wed May  4 19:05:55 UTC 2016
     [exec] With timeout 240, running:  docker  "build" "-t" "whisk/cli" "/home/vagrant/openwhisk/build/tools/cli"
     [exec] docker process has pid 13036
     [exec] 13036 pts/0    00:00:00 bash
     [exec] 13036 is still in the ps output. Must still be running.
     [exec] Cannot connect to the Docker daemon. Is the docker daemon running on this host?
     [exec] Docker process 13036 finished in time with status 1.
     [exec] Sleeping for 60 seconds to allow recovery (even though it wasn't killed)
     [exec] Cannot connect to the Docker daemon. Is the docker daemon running on this host?
     [exec] Done sleeping
     [exec] Wed May  4 19:07:00 UTC 2016
     [exec] With timeout 240, running:  docker  "build" "-t" "whisk/cli" "/home/vagrant/openwhisk/build/tools/cli"
     [exec] docker process has pid 13827
     [exec] 13827 pts/0    00:00:00 bash
     [exec] 13827 is still in the ps output. Must still be running.
     [exec] Docker process 13827 finished in time with status 1.
     [exec] Sleeping for 60 seconds to allow recovery (even though it wasn't killed)
     [exec] Done sleeping
     [exec] Wed May  4 19:08:05 UTC 2016
     [exec] With timeout 240, running:  docker  "build" "-t" "whisk/cli" "/home/vagrant/openwhisk/build/tools/cli"
     [exec] docker process has pid 14529
     [exec] 14529 pts/0    00:00:00 bash
     [exec] 14529 is still in the ps output. Must still be running.
     [exec] Cannot connect to the Docker daemon. Is the docker daemon running on this host?
     [exec] Docker process 14529 finished in time with status 1.
     [exec] Sleeping for 60 seconds to allow recovery (even though it wasn't killed)
     [exec] Cannot connect to the Docker daemon. Is the docker daemon running on this host?
     [exec] Done sleeping
     [exec] Wed May  4 19:09:10 UTC 2016
     [exec] With timeout 240, running:  docker  "build" "-t" "whisk/cli" "/home/vagrant/openwhisk/build/tools/cli"
     [exec] docker process has pid 15231
     [exec] 15231 pts/0    00:00:00 bash
     [exec] 15231 is still in the ps output. Must still be running.
     [exec] Docker process 15231 finished in time with status 1.
     [exec] Sleeping for 60 seconds to allow recovery (even though it wasn't killed)
     [exec] Done sleeping
     [exec] dockerWithRetry exiting with 1

BUILD FAILED
/home/vagrant/openwhisk/build.xml:55: The following error occurred while executing this line:
/home/vagrant/openwhisk/tools/cli/build.xml:44: exec returned: 1

Total time: 6 minutes 58 seconds
```

**Comment by rabbah** [comment]
It's not ideal that the CLI is built as part of deploy (@markusthoemmes will want to weigh in here) but there is a container build step in the current setup and this looks like it timed out also. I think this is related to your other issue - and perhaps bumping the default timeout across the board is best.

**Comment by csantanapr** [comment]
@rabbah I ran `ant reploy` again and still failed
Then I did `docker ps` and I got the error Cannot connect to the Docker daemon. Is the docker daemon running on this host?

So I `logout` and `vagrant reload` and `vagrant ssh` and now docker host is running and `docker ps` worked.

So it looks like added a verifcation step in the README for the user to check that `docker ps` works before running `ant deploy`

The root cause would be to figure out why docker host is not running after a failure in build or deploy, but I was not sure which one was the culprit
